src/coreclr/nativeaot/Runtime/riscv64/InteropThunksHelpers.S

```diff
@@ -25,8 +25,9 @@
         // If using TLS emulation, fetch the TLS data block address
         GETTHUNKDATA_ETLS_9
 #else
-        // Inline function to get TLS variable
+        mv t2, a0
         INLINE_GET_TLS_VAR t0, C_FUNC(tls_thunkData)
```

Comment:
I'd try to use the amd64 pattern where `INLINE_GET_TLS_VAR` has a `trashReg` parameter, so we can move the `a0` register saving there.

Reply:
`INLINE_GET_TLS_VAR` doesn't have trashReg arg in amd64:
https://github.com/dotnet/runtime/blob/f196875c5d0441364f6538d2dd8fc4e3eb1ff108/src/coreclr/nativeaot/Runtime/unix/unixasmmacrosamd64.inc#L264

Reply:
It does in the Windows version: https://github.com/dotnet/runtime/blob/f196875c5d0441364f6538d2dd8fc4e3eb1ff108/src/coreclr/nativeaot/Runtime/amd64/AsmMacros.inc#L222

(so much for consistency, the CFI macros also differ in negative/positive offsets between platforms)

Reply:
Ah, I added it at the last position, not pedantically it's still not consistent but call site looks cleaner. 😅